File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.java
import java.util.*;
class Solution {
    public boolean solution(String[] phone_book) {
        boolean answer = true;
        Arrays.sort(phone_book);
        for(int i=0;i<phone_book.length-1;i++){
            String now = phone_book[i];
            String temp = phone_book[i+1];
                int max_size = Math.min(now.length(),temp.length());
                String pre = temp.substring(0,max_size);
                //System.out.println(now+" "+pre);
                if(now.equals(pre)) {
                    answer = false;
                    break;
                }
            if(!answer) break;
        }
        return answer;
    }
}

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.js
function solution(phone_book) {
    var answer = true;
    phone_book.sort()
    
    for(let i=0;i<phone_book.length;i++){
        const now = phone_book[i]
        for(let j=i+1;j<phone_book.length;j++){
            let check = false
            let con = false
            const next = phone_book[j]
            for(let w=0;w<now.length;w++){
                if(w===0&&now[w]!==next[w]){
                    check = true;
                    break;
                }
                if(now[w]!==next[w]){
                    if(now[w]<next[w]) check = true
                    con = true
                    break;
                }
            }
            if(check) break;
            if(con) continue;
            return false
        }
    }
    
    return answer;
}